### PR TITLE
Borrow indexing APIs from NumPy

### PR DESCRIPTION
**Pull request by kmaehashi**

### CODE REVIEW

File: cupy/__init__.py [hunk 1]
```diff
@@ -436,6 +436,14 @@ def result_type(*arrays_and_dtypes):
 
 from cupy._indexing.iterate import flatiter  # NOQA
 
+# Borrowed from NumPy
+from numpy import get_array_wrap  # NOQA
+from numpy import index_exp  # NOQA
+from numpy import ndindex
+from numpy import s_  # NOQA
+from numpy import mask_indices  # NOQA
+from numpy import tril_indices  # NOQA
```
Comment (toslunar) [inline cupy/__init__.py:445]: `mask_indices` and `tril_indices` return tuple of arrays.  Maybe GPU arrays are expected.

Comment (kmaehashi) [reply cupy/__init__.py:445]: Thanks! Yes that was the thing what I was not confident and that's why this is marked WIP. Will fix tomorrow!

Comment (kmaehashi) [reply cupy/__init__.py:445]:
```suggestion
```

File: cupy/__init__.py [hunk 2]
```diff
@@ -436,6 +436,12 @@ def result_type(*arrays_and_dtypes):
 
 from cupy._indexing.iterate import flatiter  # NOQA
 
+# Borrowed from NumPy
+from numpy import get_array_wrap  # NOQA
+from numpy import index_exp  # NOQA
+from numpy import ndindex
```
Comment (kmaehashi) [inline cupy/__init__.py:442]:
```suggestion
from numpy import ndindex  # NOQA
```

**Review by toslunar — APPROVED** [review]

File: cupy/__init__.py (hunk 1 above)
Comment (kmaehashi) [reply cupy/__init__.py]: Added them to #6078

**Pull request merged**